### PR TITLE
Basic models needed for adding doctors

### PR DESCRIPTION
**Pull request by howinator**
I have included three models that should provide the basics for submitting 
doctors. I have included the commit messages below. I used this code
to succesfully migrate in Django.
 Changes:

 5144a61 (Howie Benefiel, 5 hours ago)
    Migration ran succesfully for this model. This is cleared for launch.

 ca106e4 (Howie Benefiel, 5 hours ago)
    First version of the models required for representing the doctors. Two
    sub-models were needed: Address and Specialty. There is a one-to-many
    relationship between Address and Doctor while there is a many-to-many
    replationship between Doctor and Specialty.

 d436c30 (Howie Benefiel, 19 hours ago)
    Started working on doctors and addresses models

### CODE REVIEW

**Comment by howinator** [comment]
migrations/0001_initial.py was automatically created by django - it simply used my models.py file to egenrate that code

File: meddjangoproject/the_big_one/models.py [hunk 1]
```diff
@@ -1,3 +1,54 @@
...
-# Create your models here.
+
+class Address(models.Model):
+  # This table stores all the addresses for doctors
+  # TODO pretty much change all this to use local flavor 
+
+  street_address = models.CharField(max_length = 250)
+  city = models.CharField(max_length = 150)
+  state = models.CharField(max_length = 2)
+  zip_code = models.CharField(max_length = 5)
```
Comment (sarangj) [inline meddjangoproject/the_big_one/models.py]: Are we going to allow zip codes of the flavor xxxxx-xxxx?

File: meddjangoproject/the_big_one/models.py [hunk 2]
```diff
@@ -1,3 +1,54 @@
...
+  # street_number and street_name will be optional fields that can be used if
+  # we need t decompose the address into its arts
```
Comment (sarangj) [inline meddjangoproject/the_big_one/models.py]: nittiest of nits: we need to*

File: meddjangoproject/the_big_one/models.py [hunk 3]
```diff
@@ -1,3 +1,54 @@
...
+  street_number = models.CharField(max_length = 15, blank = True)
+  street_name = models.CharField(max_length = 200, blank = True)
+  is_active = models.BooleanField(
+      verbose_name = 'is address actively being used', default = True)
```
Comment (sarangj) [inline meddjangoproject/the_big_one/models.py]: I think the usual convention for multiline args is something like

  is_active = models.BooleanField(
      verbose_name = 'is address actively being used',
      default = True
  )

File: meddjangoproject/the_big_one/models.py [hunk 4]
```diff
@@ -1,3 +1,54 @@
...
+  user_submitted = models.BooleanField(default = False)
+  # Two time stamp fields
+  # auto_now_add time stamps when it is first created
+  date_entered = models.DateTimeField(
```
Comment (sarangj) [inline meddjangoproject/the_big_one/models.py]: Same as the above regarding styling

File: meddjangoproject/the_big_one/models.py [hunk 5]
```diff
@@ -1,3 +1,54 @@
...
+      verbose_name = 'date+time address entered', auto_now_add = True)
+  # auto_now time stamps when the field is saved
+  date_last_updated = models.DateTimeField(
```
Comment (sarangj) [inline meddjangoproject/the_big_one/models.py]: etc.

File: meddjangoproject/the_big_one/models.py [hunk 6]
```diff
@@ -1,3 +1,54 @@
...
+  # doctor can have multiple specialties and specialties can belong to 
+  # multiple doctors therefore, many-to-many relationship
+  specialties = models.ManyToManyField(Specialty)
+  email_address = models.EmailField()
+  # TODO change this to use localflavor
+  phone_number = models.CharField(max_length = 9)
+  # This field should be automatically calculated by a script nightly
+  average_price = models.DecimalField(max_digits = 10, decimal_places = 2)
+
+  def __str__(self):
```
Comment (sarangj) [inline meddjangoproject/the_big_one/models.py]: This is neat.

File: meddjangoproject/the_big_one/models.py (hunk 1 above)
Comment (howinator) [inline meddjangoproject/the_big_one/models.py]: yeah I'll make this change.

File: meddjangoproject/the_big_one/models.py (hunk 6 above)
Comment (howinator) [inline meddjangoproject/the_big_one/models.py]: you're neat

File: meddjangoproject/the_big_one/models.py (hunk 2 above)
Comment (howinator) [inline meddjangoproject/the_big_one/models.py]: and parts*

File: meddjangoproject/the_big_one/models.py [hunk 7]
```diff
@@ -1,3 +1,57 @@
...
+  city = models.CharField(max_length = 150)
+  state = models.CharField(max_length = 2)
+  zip_code = models.CharField(max_length = 10)
+  # street_number and street_name will be optional fields that can be used if
+  # we need to decompose the address into its parts
+  street_number = models.CharField(max_length = 15, blank = True)
+  street_name = models.CharField(max_length = 200, blank = True)
+  is_active = models.BooleanField(
+      verbose_name = 'is address actively being used', 
+      default = True)
```
Comment (sarangj) [inline meddjangoproject/the_big_one/models.py]: I think the paren goes on the next line typically :)

```python
    default = True
)
```

File: meddjangoproject/the_big_one/models.py [hunk 8]
```diff
@@ -1,3 +1,57 @@
...
+  user_submitted = models.BooleanField(default = False)
+  # Two time stamp fields
+  # auto_now_add time stamps when it is first created
+  date_entered = models.DateTimeField(
+      verbose_name = 'date+time address entered', 
+      auto_now_add = True)
```
Comment (sarangj) [inline meddjangoproject/the_big_one/models.py]: Same

File: meddjangoproject/the_big_one/models.py [hunk 9]
```diff
@@ -1,3 +1,57 @@
...
+  # auto_now time stamps when the field is saved
+  date_last_updated = models.DateTimeField(
+      verbose_name = 'date+time last changed', 
+      auto_now = True)
```
Comment (sarangj) [inline meddjangoproject/the_big_one/models.py]: Etc.

**Pull request merged**